File: his/his/report/sales_report/sales_report.js
// Copyright (c) 2022, Anfac Tech and contributors
// For license information, please see license.txt
/* eslint-disable */

// frappe.db.get_value('Defaults', {user: frappe.session.user , document : "Account"}, 'value')
// 			.then(r => {		
frappe.query_reports["Sales Report"] = {
	"filters": [
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			default:frappe.datetime.now_date(),
			reqd: 1
		},
		{
			fieldname: "to",
			label: __("To Date"),
			fieldtype: "Date",
			default:frappe.datetime.now_date(),
			reqd: 1
		},
		

	]
};
// })
//default :r.message.value

File: his/his/report/sales_report/sales_report.py
# Copyright (c) 2022, Anfac Tech and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
	
	return get_columns(), get_data(filters)

def get_data(filters):
	
	_from ,to  = filters.get('from_date'), filters.get('to') 
	
	data = frappe.db.sql(f"""
	select 
	posting_date,
	name, 
	customer_name ,
	net_total,
	paid_amount,
	discount_amount,
	outstanding_amount,
	
	owner  

from `tabSales Invoice`
where posting_date between "{_from}" and "{to}"  and docstatus = 1
 ;""")
	return data
def get_columns():
	return [

		"Date: Date:120",
		"Reciept No:Link/Sales Invoice:220",
		"Customer Name:Link/Customer:200", 
		"Amount:Currency:110",
		
		"Piad:Currency:110",
		"Discount:Currency:110",
		"Unpaid:Currency:110",
		"User:Link/User:220",

	
		
	]

